docs/root/version_history/current.rst

```diff
@@ -28,6 +28,7 @@ Minor Behavior Changes
   in the environment.
 * router: added transport failure reason to response body when upstream reset happens. After this change, the response body will be of the form `upstream connect error or disconnect/reset before headers. reset reason:{}, transport failure reason:{}`.This behavior may be reverted by setting runtime feature `envoy.reloadable_features.http_transport_failure_reason_in_body` to false.
 * router: now consumes all retry related headers to prevent them from being propagated to the upstream. This behavior may be reverted by setting runtime feature `envoy.reloadable_features.consume_all_retry_headers` to false.
+* router_check_tool: `request_header_fields`, `response_header_fields` config deprecated in favor of `request_header_matches`, `response_header_matches`.
```

Comment:
I think this needs to go to the "Deprecated" section below?

Reply:
Sure. Is it ok to hard-switch the accepted configs then? I might have to accept the old format or the new format and then pull the old format after the next minor release.

Reply:
IMO it's not OK to hard switch. Can we do a normal deprecation and support both until the next release?

Reply:
I brought back the code to process `request_header_fields` and `response_header_fields`.